packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -121,20 +124,23 @@ export abstract class AbstractBackgroundJobProcessor<
   ) {
     this.config = config
     this.factory = dependencies.bullmqFactory
-    this.redis = dependencies.redis
+    this.redis = new Redis(config.redisConfig)
     this.transactionObservabilityManager = dependencies.transactionObservabilityManager
     this.logger = dependencies.logger
     this.errorReporter = dependencies.errorReporter
     this._spy = config.isTest ? new BackgroundJobProcessorSpy() : undefined
+    this.runningPromises = []
   }
 
-  public static async getActiveQueueIds(redis: Redis): Promise<string[]> {
-    await redis.zremrangebyscore(
+  public static async getActiveQueueIds(redisConfig: RedisConfig): Promise<string[]> {
```

Comment:
🟡 This method is used from bull-board to retrieve the list of queues for the given connection, this method is called with an interval to refresh the on UI, wondering if we can still pass the Redis object instead of the config to not have to create a new connection with each call

Reply:
Just as an idea, maybe instead removing it we can use it as `redisWithoutPrefix` we have in `start` method

Reply:
I added the possibility to pass either a connection or the configuration to fit both use cases 🙏